packages/data-service/src/data-service.spec.ts

```diff
@@ -1477,6 +1478,65 @@ describe('DataService', function () {
         expect(uuid).to.deep.equal(keyDoc._id);
       });
     });
+
+    describe('#explainAggregate', function () {
+      const initialAbortController = global.AbortController;
+      before(function () {
+        (global as any).AbortController = AbortController;
+      });
+      after(function () {
+        global.AbortController = initialAbortController;
+      });
+      it('returns an explain object', async function () {
+        const explain = await dataService.explainAggregate(
+          testNamespace,
+          [
+            {
+              $match: {
+                a: 1,
+              },
+            }
+          ],
+          {},
+          {}
+        );
+        expect(explain).to.be.an('object');
+      });
+      it('returns an explain object - cancellable', function (done) {
+        const abortController = new AbortController();
+        const abortSignal = abortController.signal;
+        const pipeline = [
+          {
+            $addFields: {
+              lazy: {
+                $function: {
+                  body: `function () {
+                    return sleep(1000);
+                  }`,
+                  args: [],
+                  lang: "js"
+                }
+              }
+            },
+          }
+        ];
+        const executionOptions = {
+          abortSignal,
+        };
+
+        dataService.explainAggregate(
+          testNamespace,
+          pipeline,
+          {},
+          executionOptions as any
+        )
+          .catch((error) => {
```

Comment:
What about we use the `async/await` for of this and remove these callback style tests?

```
const error = dataService.explainAggregate(...).catch(err => err);
expect(error).to.be.instance.of(Error);
expect(dataService.isOperationCancelledError(error)).to.true;
```